monitoring/google/cloud/monitoring/_dataframe.py

```diff
@@ -0,0 +1,143 @@
+# Copyright 2016 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Time series as :mod:`pandas` dataframes."""
+
+import itertools
+
+from google.cloud.monitoring_v3.types import TimeSeries
+
+TOP_RESOURCE_LABELS = (
+    'project_id',
+    'aws_account',
+    'location',
+    'region',
+    'zone',
+)
+
+
+def _extract_header(time_series):
+    """Return a copy of time_series with the points removed."""
+    return TimeSeries(
+        metric=time_series.metric,
+        resource=time_series.resource,
+        metric_kind=time_series.metric_kind,
+        value_type=time_series.value_type
+    )
+
+
+def _extract_labels(time_series):
+    """Build the combined resource and metric labels, with resource_type."""
+    labels = {'resource_type': time_series.resource.type}
+    labels.update(time_series.resource.labels)
+    labels.update(time_series.metric.labels)
+    return labels
+
+
+def _extract_value(typed_value):
+     """Extract the value from a TypedValue."""
+     value_type = typed_value.WhichOneof('value')
+     return typed_value.__getattribute__(value_type)
+
+
+def _build_dataframe(time_series_iterable,
+                     label=None, labels=None):  # pragma: NO COVER
+    """Build a :mod:`pandas` dataframe out of time series.
+
+    :type time_series_iterable:
+        iterable over :class:`~google.cloud.monitoring_v3.types.TimeSeries`
+    :param time_series_iterable:
+        An iterable (e.g., a query object) yielding time series.
+
+    :type label: str
+    :param label:
+        (Optional) The label name to use for the dataframe header. This can be
+        the name of a resource label or metric label (e.g.,
+        ``"instance_name"``), or the string ``"resource_type"``.
+
+    :type labels: list of strings, or None
+    :param labels:
+        A list or tuple of label names to use for the dataframe header.
+        If more than one label name is provided, the resulting dataframe
+        will have a multi-level column header.
+
+        Specifying neither ``label`` or ``labels`` results in a dataframe
+        with a multi-level column header including the resource type and
+        all available resource and metric labels.
+
+        Specifying both ``label`` and ``labels`` is an error.
+
+    :rtype: :class:`pandas.DataFrame`
+    :returns: A dataframe where each column represents one time series.
+    """
+    import pandas   # pylint: disable=import-error
+
+    if labels is not None:
```

Comment:
A better way to go about this might be to coerce this into a sequence, always:

```python
if label is not None:
    if labels:
        raise ValueError('Cannot specify both `label` and `labels`.')
    labels = (label,)
```

Reply:
Done